### PR TITLE
simplified cses-1144 code

### PR DESCRIPTION
**Pull request by RedBlueBird**
_If any of the below doesn't apply to this Pull Request, mark the checkbox as done._

- [ ] I have tested my code.
- [ ] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [ ] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which include the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making them.

### CODE REVIEW

**Comment by RedBlueBird** [comment]
today I have learned you can create more than one pull request at the same time in Github

File: solutions/cses-1144.mdx [hunk 1]
```diff
@@ -19,64 +19,60 @@ TLE.
...
+ll bit[MAX]; // binary-indexed tree
+vector<int> vals; // values sorted for coordinate compression
 int n, q;
 
-void upd (int i, int val) {
-	for (; i <= MX; i += i&(-i)) {
-		bit[i] += val;
+void add(int value, int amount) {
+	int index = upper_bound(vals.begin(), vals.end(), value) - vals.begin();
+	for (; index <= MAX; index += index&(-index)) {
```
Comment (bqi343) [inline solutions/cses-1144.mdx:30]: this goes out of bounds. also, the parentheses around `(-index)` are not necessary

File: solutions/cses-1144.mdx [hunk 2]
```diff
@@ -19,64 +19,60 @@ TLE.
...
-ll query(int x) {
-	int ind = upper_bound(vals.begin(), vals.end(), x) - vals.begin();
-	return sum(ind);
+	return total;
 }
 
 int main() {
 	cin >> n >> q;
-	vector<int> ar(n);
+	vector<int> salaries(n); // salaries of n employees
```
Comment (bqi343) [inline solutions/cses-1144.mdx:46]: > // salaries of n employees

not necessary

File: solutions/cses-1144.mdx [hunk 3]
```diff
@@ -381,47 +172,37 @@ Just use indexed set!
...
-Tree<pii> o;
+Tree<pii> indexed_set;
 
 int main() {
-	ios_base::sync_with_stdio(0); cin.tie(0);
+	ios_base::sync_with_stdio(0); 
+	cin.tie(0);
 	cin >> n >> q;
-	FOR(i,1,n+1) {
+	for (i = 1; i < n + 1; ++i) {
```
Comment (bqi343) [inline solutions/cses-1144.mdx:187]: this doesn't compile

**Review by bqi343 — CHANGES_REQUESTED** [review]

File: solutions/cses-1144.mdx [hunk 4]
```diff
@@ -90,282 +86,77 @@ limit.
...
-		if (t == '!') vals[b] = 0;
+	setIO(); 
+	cin >> N >> Q;
+	vector<int> salaries(N); // salaries of n employees
+	vector<array<int, 3>> queries; // store the input queries offline
+
+	for (int i = 0; i < N; i++){
+		cin >> salaries[i];
+	}
+	for (auto& t: salaries) {
```
Comment (bqi343) [inline solutions/cses-1144.mdx:132]: no need for `&` here.

File: solutions/cses-1144.mdx [hunk 5]
```diff
@@ -19,64 +19,60 @@ TLE.
 using namespace std;
 using ll = long long;
 
-const int MX = 4e5 + 5;
+const int MAX = 4e5 + 5;
 
-ll bit[MX];
-vector<int> vals;
+ll bit[MAX]; // binary-indexed tree
```
Comment (SansPapyrus683) [inline solutions/cses-1144.mdx:24]:
```suggestion
ll bit[MAX];  // binary-indexed tree
```same for other comments

File: solutions/cses-1144.mdx [hunk 6]
```diff
@@ -90,282 +86,77 @@ limit.
...
-	trav(p,P) ad(p,1);
-	trav(u,rec) {
-		u[1] --;
-		if (u[0] == 0) {
-			ad(P[u[1]],-1);
-			P[u[1]] = u[2];
-			ad(P[u[1]],1);
-		} else ps(query(u[2])-query(u[1]));
+	int count = 0; 
+	for (auto& t: vals) {
```
Comment (SansPapyrus683) [inline solutions/cses-1144.mdx:145]: please don't use `auto`

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/cses-1144.mdx (hunk 6 above)
Comment (RedBlueBird) [reply solutions/cses-1144.mdx]: what should I change it to?

Comment (SansPapyrus683) [reply solutions/cses-1144.mdx]: use the actual type name please

File: solutions/cses-1144.mdx [hunk 7]
```diff
@@ -90,282 +87,77 @@ limit.
...
-
-BIT<int,1000005> B;
-
-map<int,int> vals;
-int N,Q;
-
-void ad(int a, int b) {
-	int ind = vals[a];
-	B.upd(ind,b);
+	else index = it->second-1;
```
Comment (SansPapyrus683) [inline solutions/cses-1144.mdx:109]: braces

File: solutions/cses-1144.mdx [hunk 8]
```diff
@@ -90,282 +87,77 @@ limit.
...
+	int total = 0;
+	for (; index; index -= index&(-index)) {
```
Comment (SansPapyrus683) [inline solutions/cses-1144.mdx:111]: spaces between operators

File: solutions/cses-1144.mdx [hunk 9]
```diff
@@ -90,282 +87,79 @@ limit.
...
-			ad(P[u[1]],1);
-		} else ps(query(u[2])-query(u[1]));
+	int count = 0; 
+	for (auto t: vals) {
+		t.second = ++count;
+	}
+	for (auto p: salaries){
+		add(p, 1);
+	}
+	for (auto query: queries) {
```
Comment (SansPapyrus683) [inline solutions/cses-1144.mdx:152]: please don't use `auto`
also do `for (i : y)` instead of `for (i: y)`

**Comment by SansPapyrus683** [comment]
jeff can u

**Comment by SansPapyrus683** [comment]
nonononono

File: solutions/cses-1144.mdx (hunk 9 above)
Comment (SansPapyrus683) [reply solutions/cses-1144.mdx]:
```suggestion
for (const pair<int, int>& query : queries) {
```

**Pull request closed without merging**